x/gamm/README.md

```diff
@@ -86,8 +86,12 @@ All tokens are swapped using a multi-hop mechanism. That is, all swaps
 are routed via the most cost-efficient way, swapping in and out from
 multiple pools in the process.
 
+When a trade consists of just two OSMO-included routes during a single transaction,
+the swap fees on each hop would be automatically halved.
+
```

Comment:
Thoughts on adding an example here?

Also, does it matter if we half the `swapFee` at the end (totalfee / 2) vs halving it per pool swap with usmo?

Reply:
Yes, in fact `totalfee` will be reduced by  less than by 50%, depending on various pool states. On test introduced in that PR it will be around 25%, cuz we create a pool with low liquidity.  In general final percentage will grow up to 50% the higher liquidity is.
I just made it as in proposal description(see forum) and sunny first implementation. I think it more elegant and stable solution, rather than trying to make it half for all cases.

Reply:
added example in re-base commit